topostats/plotting.py

```diff
@@ -1,650 +1,351 @@
+        """
+        self.df = df if df is not None else pd.read_csv(csv_file)
+        self.stat_to_sum = stat_to_sum
+        self.molecule_id = molecule_id
+        self.image_id = image_id
+        self.hist = hist
+        self.bins = bins
+        self.stat = stat
+        self.kde = kde
+        self.cut = cut
+        self.figsize = figsize
+        self.alpha = alpha
+        self.palette = palette
+        self.file_ext = file_ext
+        self.output_dir = Path(output_dir)
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+        self.var_to_label = var_to_label
+        self.melted_data = None
+        self.summary_data = None
+        self.label = None
+        self.sns_melt_data()
+        self.set_palette()
+        self._set_label(self.stat_to_sum)
+
+    def _setup_figure(self):
+        """Setup Matplotlib figure and axes."""
+        fig, ax = plt.subplots(1, 1, figsize=self.figsize)
+        return fig, ax
+
+    def _outfile(self, plot_suffix: str) -> str:
+        """Generate the output file name with the appropriate suffix.
+
+        Parameters
+        ----------
+        plot_suffix: str
+            The suffix to append to the output file.
+
+        Returns
+        -------
+        str:
+            Concanenated string of the outfile and plot_suffix.
+        """
+        return f"{self.stat_to_sum}_{plot_suffix}"
+
+    def sns_plot(self) -> None:
+        """Plot the distribution of one or more statistics as either histogram, kernel density estimates or both. Uses
+        base Seaborn."""
+        fig, ax = self._setup_figure()
+        if self.hist and not self.kde:
+            outfile = self._outfile("hist")
+            sns.histplot(data=self.melted_data, x="value", bins=self.bins, stat=self.stat, hue=self.image_id)
+        if self.kde and not self.hist:
+            outfile = self._outfile("kde")
+            sns.kdeplot(data=self.melted_data, x="value", hue=self.image_id)
+        if self.hist and self.kde:
+            outfile = self._outfile("hist_kde")
+            sns.histplot(
+                data=self.melted_data,
+                x="value",
+                bins=self.bins,
+                stat=self.stat,
+                hue=self.image_id,
+                kde=True,
+                kde_kws={"cut": self.cut},
+            )
+        plt.ticklabel_format(axis="both", style="sci", scilimits=(-3, 3))
+        plt.title(self.label)
+        self.set_xlim()
+        self.save_plot(outfile)
+        return fig, ax
+
+    def sns_violinplot(self) -> None:
+        """Violin plot of data."""
+        fig, ax = self._setup_figure()
+        sns.violinplot(data=self.melted_data, x=self.image_id, y="value", hue=self.image_id, alpha=self.alpha)
+        plt.title(self.label)
+        plt.xlabel("Image")
+        plt.ylabel(self.label)
+        outfile = self._outfile("violin")
+        self.save_plot(outfile)
+        return fig, ax
+
+    # def sns_jointplot(self, var1: str, var2: str) -> None:
```

Comment:
`jointplot` was one of the features @Jean-Du had in the previous version. I'm undecided as to how easy/sensible it would be to implement given the possiblilty of all pair-wise combinations from a list of selected variables or how one might configure it more manually (thinking we don't want an overly complicated configuration system).

I've noted this in the **ToDo (maybe)** section of the PR along with other things, some of which may become issues/tasks to be completed.